lebedigital/raw_data_processing/youngs_modulus_data/emodul_metadata_extraction.py

```diff
@@ -73,60 +70,77 @@ def extract_metadata_emodulus(rawDataPath,specimen_file,mix_file):
+        metadata['MeasurementDuration'] = float(replace_comma(serviceInformation[10][2]))
+
+        # operator name - This data has no placeholder yet.
+        metadata['tester_name'] = serviceInformation[2][1] 
+
+        # remarks - This data has no placeholder yet.
         metadata['remark'] = serviceInformation[4][1]
 
-        # weight
+        # set experiment lab location to BAM
+        metadata['Lab'] = 'BAM'
+
+        # set Compression and Transducer Column
+        metadata['CompressionColumn'] = 0
+        metadata['TransducerColumn'] = [1,2,3]
+
+        # set paths
+        metadata['ProcessedFile'] = os.path.join('../usecases/MinimumWorkingExample/emodul/processed_data') # path to csv file with values extracted by emodul_generate_processed_data.py
+        metadata['RawDataFile'] = os.path.join(rawDataPath,specimen_file) # path to specimen.dat
+        try:
+            with open(str(rawDataPath)+'/'+ str(mix_file), encoding="utf8", errors='ignore') as mix_data:
+                lines = mix_data.readlines()
+                lines = lines[0].strip()
+                dataPath = Path(rawDataPath).parents[1]
+                metadata['MixDataFile']= os.path.join(dataPath, "Mischungen", lines)
+        except:
+            metadata['MixDataFile'] = None
+
+
+        ###########  D A T A   A B O U T    S P E C I M E N #######
+
+        # name of specimen 
+        metadata['SpecimenName'] = serviceInformation[3][1] 
+
+        # set specimen age to 28 days
+        metadata['SpecimenAge'] = 28.0
+
+        # weight - This data has no placeholder yet.
         weight = float(replace_comma(serviceInformation[5][1]))
         metadata['weight'] = weight
 
-        # set weight unit, only a rough approximation
-        # should be included in the input data in the long run
-        if metadata['weight'] > 1000:
-            metadata['weight_unit'] = 'g'
-        elif metadata['weight'] < 10:
-            metadata['weight_unit'] = 'kg'
-        else:
-            raise Exception('Unexpected value of weight')
-
         # set size of specimen
-        metadata['diameter'] = float(replace_comma(serviceInformation[6][1]))
-        metadata['length'] = float(replace_comma(serviceInformation[7][1]))
-
-        if metadata['diameter'] > metadata['length']:
+        metadata['SpecimenDiameter'] = float(replace_comma(serviceInformation[6][1])) #diameter
+        metadata['SpecimenLength'] = float(replace_comma(serviceInformation[7][1])) #length
+        if metadata['SpecimenDiameter'] > metadata['SpecimenLength']:
             dir_name = metadata['experimentName']
             raise Exception(f'Diameter is larger then length, please fix the mistake in {dir_name}')
 
-        # set units, assuming length and diameter have the same
-        if metadata['length'] > 100:
+        # set units, assuming length and diameter have the same - This data has no placeholder yet.
+        if metadata['SpecimenLength'] > 100:
```

Comment:
Could we just change the files and assume everything is in m (or mm)? This way, it is quite tricky to find bugs in future data.

Reply:
I think Erik raised an issue with including the units in the metadata, so I guess using a single unit would be more consistent

Reply:
Jörg refers to: https://github.com/BAMresearch/LebeDigital/issues/153
I am not sure if using a single unit is the best option in the long run, but for now I am happy if it is included anywhere.
I thought it might be easier and more flexible to give every entry a unit. This includes then automatically the dimensionality of the value. Otherwise we would need to know that kind of unit is connected to which value.